ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -78,6 +78,61 @@ objects:
 
     containers:
 
+    - name: lease
+      image: ci/boskoscli:latest
+      terminationMessagePolicy: FallbackToLogsOnError
+      resources:
+        requests:
+          cpu: 10m
+          memory: 10Mi
+        limits:
+          memory: 200Mi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /tmp/cluster
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: CLUSTER_TYPE
+        value: ${CLUSTER_TYPE}
+      - name: CLUSTER_NAME
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        if [[ "${CLUSTER_TYPE:-}" != "aws" ]]; then
+          echo "[WARNING] No resource leasing is defined for \"${CLUSTER_TYPE}\" clusters"
+          exit 0
+        fi
+
+        resource="$( boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" acquire --type aws-quota-slice --state clean --target-state leased --timeout 30m )"
+        touch /tmp/shared/leased
+
+        function release() {
+            local resource_name; resource_name="$( jq .name <<<"${resource}" )"
+            boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" release --name "${resource_name}" --target-state clean
+        }
+        trap release EXIT
+
+        boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" heartbeat --resource "${resource}" --period 30s --timeout 4h &
+
+        while true; do
+          if [[ -f /tmp/shared/exit ]]; then
+            echo "Another process exited" 2>&1
+            exit 0
+          fi
+
+          sleep 15 & wait
```

Comment:
A bare `wait` here will catch the earlier, backgrounded `boskosctl` heartbeat as well, but you only want to wait on the `sleep`.  How about:

```sh
sleep 15 &
wait $!
```